### DISCUSSION TITLE
Running lighthouse for cached pages

### DISCUSSION THREAD

**Discussion by chaitanya1248** [GoogleChrome/lighthouse#15545] [Q&A] [answered]
We use lighthouse in a MEAN stack app to record and store the performance metrics.

The ask is to compare the lighthouse performance of a page with cache and without cache.

We use puppeteer launcher and run the lighthouse.

Is there a recommended flow of running lighthouse to get performance comparision for a page with cache and without cache?

**Comment by connorjclark** [comment, marked as answer]
You want the `--disable-storage-reset` / [`disableStorageReset`](https://github.com/GoogleChrome/lighthouse/blob/main/types/lhr/settings.d.ts#L71C3-L71C22) setting. Run first without that, then again with that setting. You should use the same puppeteer instance so that the cache persists (when LH launches from the CLI, it uses a fresh profile, so no good).

**Reply by chaitanya1248** [reply]
Sure. I'll load the page and provide that puppeteer instance to LH. Thank you.
